dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPIImpl.java

```diff
@@ -307,15 +322,33 @@ public List<FileAsset> fromContentlets(final List<Contentlet> contentlets) {
 
 	}
 
-	public List<IFileAsset> fromContentletsI(final List<Contentlet> contentlets) {
+	public List<IFileAsset> fromContentletsI(final List<Contentlet> contentlets) throws DotDataException, DotSecurityException {
 		final List<IFileAsset> fileAssets = new ArrayList<>();
 		for (Contentlet con : contentlets) {
-			fileAssets.add(fromContentlet(con));
+			if (con.isDotAsset()) {
+
+				fileAssets.add(transformDotAsset(con));
+			} else {
+				fileAssets.add(fromContentlet(con));
+			}
 		}
 		return fileAssets;
 
 	}
 
+	private FileAsset transformDotAsset(Contentlet con) throws DotDataException, DotSecurityException {
+		con.setProperty(FileAssetAPI.BINARY_FIELD, Try.of(()->con.getBinary("asset")).getOrNull());
+
+		FileAsset fileAsset = FileViewStrategy.convertToFileAsset(con, this);
+
+		final HttpServletRequest request = HttpServletRequestThreadLocal.INSTANCE.getRequest();
```

Comment:
Might be worth to add a null check null for the request here? Just in case this method is called from a separate thread or job, so there isn't. a current request set.

Reply:
Good point @dsolistorres . I've ran into situations in which getting the current request like this returns a `null`.